File: packages/witness/src/bin/testivai.ts
#!/usr/bin/env node

import { Command } from 'commander';
import chalk from 'chalk';
import { initCommand } from '../commands/init';
import { runCommand } from '../commands/run';
import { witnessCommand } from '../commands/capture';
import { approveCommand } from '../commands/approve';
import { mergeCapturesCommand } from '../commands/merge-captures';
import { reportCommand } from '../commands/report';

const packageJson = require('../../package.json');

const program = new Command();

// Display banner
const showBanner = () => {
  console.log();
  console.log(chalk.cyan.bold('  TestivAI'));
  console.log(chalk.gray('  Local-first visual regression testing'));
  console.log();
};

program
  .name('testivai')
  .description('TestivAI Witness SDK - Framework-agnostic visual regression testing')
  .version(packageJson.version, '-V, --version', 'Display version number')
  .hook('preAction', () => {
    // Keep stdout clean for machine-readable output: no banner under --json
    // (or --quiet/-q). Agents parse stdout as a single JSON document.
    const quiet =
      process.argv.includes('--quiet') ||
      process.argv.includes('-q') ||
      process.argv.includes('--json');
    if (!quiet) {
      showBanner();
    }
  });

// Global options
program
  .option('-v, --verbose', 'Enable verbose output')
  .option('-q, --quiet', 'Suppress output (ideal for CI)')
  .option('--debug', 'Enable debug mode');

// Add commands
program.addCommand(initCommand);
program.addCommand(runCommand);
program.addCommand(witnessCommand);
program.addCommand(approveCommand);
program.addCommand(mergeCapturesCommand);
program.addCommand(reportCommand);

// Parse arguments
program.parse();

// Show help if no command provided
if (!process.argv.slice(2).length) {
  program.outputHelp();
}

File: ruby/lib/testivai.rb
# frozen_string_literal: true

require_relative "testivai/version"
require_relative "testivai/config"
require_relative "testivai/element_map"
require_relative "testivai/shard"
require_relative "testivai/capture"

# TestivAI — local-first visual regression testing for Ruby test suites.
#
#   require "testivai"
#
#   it "renders the homepage" do
#     visit "/"
#     Testivai.witness(page, "homepage")
#   end
#
# Then compare and build the report:
#
#   npx testivai report
module Testivai
  # Capture a snapshot. See Testivai::Capture.witness for options.
  def self.witness(session, name, **options)
    Capture.witness(session, name, **options)
  end
end
